lib/shopify_api/rest/customer.ex

```diff
@@ -121,4 +121,15 @@ defmodule ShopifyAPI.REST.Customer do
   """
   def get_orders(%AuthToken{} = auth, customer_id),
     do: Request.get(auth, "customers/#{customer_id}/orders.json")
+
+  @doc """
+  Search for customers that match a supplied query
+
+  ## Example
+
+      iex> ShopifyAPI.REST.Customer.search(auth, query_string)
+      {:ok, {"customers" => [] }}
+  """
+  def search(%AuthToken{} = auth, query_string),
+    do: Request.get(auth, "customers/search.json?query=#{query_string}")
```

Comment:
In the past we have done search filters like so https://github.com/pixelunion/elixir-shopifyapi/blob/master/lib/shopify_api/rest/asset.ex#L18